File: solutions/1603. Design Parking System/1603.cpp
// Time complexity: O(1)
// Space complexity: O(1)
class ParkingSystem {
 public:
  ParkingSystem(int big, int medium, int small) {
    count = {big, medium, small};
  }

  bool addCar(int carType) {
    return count[carType - 1]-- > 0;
  }

 private:
  vector<int> count;
};

File: solutions/1603. Design Parking System/1603.py
# Time complexity: O(1)
# Space complexity: O(1)
class ParkingSystem:
  def __init__(self, big: int, medium: int, small: int) -> None:
    self.count = [big, medium, small]

  def addCar(self, carType: int) -> bool:
    self.count[carType - 1] -= 1
    return self.count[carType - 1] >= 0